Calibration/EcalAlCaRecoProducers/plugins/BuildFile.xml

```diff
@@ -8,4 +8,15 @@
 <use   name="DataFormats/TrackReco"/>
 <use   name="CommonTools/UtilAlgos"/>
 <use   name="DataFormats/RecoCandidate"/>
+# for FED list selector
+<use   name="DataFormats/EgammaCandidates"/>
+<use   name="CalibFormats/SiStripObjects"/>
+<use   name="CalibCalorimetry/HcalAlgos"/>
+<use   name="CondFormats/EcalObjects"/>
+<use   name="CondFormats/DataRecord"/>
+<use   name="CondFormats/HcalObjects"/>
+<use   name="CondFormats/SiPixelObjects"/>
+<use   name="Geometry/CaloGeometry"/>
+<use   name="RecoEcal/EgammaCoreTools"/>
+<use   name="TrackingTools/TrackAssociator"/>
 <flags   EDM_PLUGIN="1"/>
```

Comment:
Dear David,
I'm cleaning it a bit, but a large number of includes are needed and the BuildFile will not be reduced.
Are you suggesting to replace 
use name= with  use class= ?
Can you point me to an example please?

Thanks
-- 
 							Shervin


On Mon, 1 Jun 2015, David Lange wrote:

> Date: Mon, 1 Jun 2015 09:53:58 +0200
> From: David Lange <notifications@github.com>
> Reply-To: cms-sw/cmssw
>     <reply+0045c225268f363ecc1171f055c741770e5e1d3b282d2e1d92cf000000011183d19
>     692a170ce01df364a@reply.github.com>
> To: cms-sw/cmssw <cmssw@noreply.github.com>
> Cc: shervin86 <shervin@cern.ch>
> Subject: Re: [cmssw] Electron stream plugin fix (#9256)
> 
>> @@ -8,4 +8,15 @@
>>  <use   name="DataFormats/TrackReco"/>
>>  <use   name="CommonTools/UtilAlgos"/>
>>  <use   name="DataFormats/RecoCandidate"/>
>> +# for FED list selector
>> +<use   name="DataFormats/EgammaCandidates"/>
>> +<use   name="CalibFormats/SiStripObjects"/>
>> +<use   name="CalibCalorimetry/HcalAlgos"/>
>> +<use   name="CondFormats/EcalObjects"/>
>> +<use   name="CondFormats/DataRecord"/>
>> +<use   name="CondFormats/HcalObjects"/>
>> +<use   name="CondFormats/SiPixelObjects"/>
>> +<use   name="Geometry/CaloGeometry"/>
>> +<use   name="RecoEcal/EgammaCoreTools"/>
>> +<use   name="TrackingTools/TrackAssociator"/>
>>  <flags   EDM_PLUGIN="1"/>
>
> was trying to figure out why all of these dependencies came in. Could you clean up the header file for the moved plugin to use only the incldues that it needs (which is almost none of them- use "class blah;" instead) and add the minimal set of headers to the cc file.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cms-sw/cmssw/pull/9256/files#r31405642
>